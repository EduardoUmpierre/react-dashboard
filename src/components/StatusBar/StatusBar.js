import React from 'react'
import StatusBarItem from './StatusBarItem/StatusBarItem'

const statusBar = props => (
    <div className='row'>
        <StatusBarItem theme='blue' icon='shopping-bag' text='New Orders' amount='120' />
        <StatusBarItem theme='yellow' icon='comment' text='Comments' amount='52' />
        <StatusBarItem theme='green' icon='user' text='New Users' amount='24' />
        <StatusBarItem theme='red' icon='eye' text='Page Views' amount='25.2k' />
    </div>
)

export default statusBar
