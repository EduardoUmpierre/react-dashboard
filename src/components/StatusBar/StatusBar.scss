.status-bar {
    margin-bottom: 1.250rem;
}