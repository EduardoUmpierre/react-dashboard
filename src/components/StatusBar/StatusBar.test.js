import React from 'react'
import { shallow } from 'enzyme'
import StatusBar from './StatusBar'

describe('<StatusBar />', () => {
    const widgets = {
        comments: 59,
        newOrders: 124,
        newUsers: 9,
        pageViews: 17827
    }

    it('renders correctly', () => {
        const component = shallow(<StatusBar widgets={widgets} />)

        expect(component).toMatchSnapshot()
    })
})
