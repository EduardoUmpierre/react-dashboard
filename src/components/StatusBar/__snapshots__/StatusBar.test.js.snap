// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StatusBar /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <statusBar
    widgets={
      Object {
        "comments": 59,
        "newOrders": 124,
        "newUsers": 9,
        "pageViews": 17827,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <statusBarItem
          amount={124}
          icon="shopping-bag"
          text="New Orders"
          theme="blue"
        />,
        <statusBarItem
          amount={59}
          icon="comment"
          text="Comments"
          theme="yellow"
        />,
        <statusBarItem
          amount={9}
          icon="user"
          text="New Users"
          theme="green"
        />,
        <statusBarItem
          amount={17827}
          icon="eye"
          text="Page Views"
          theme="red"
        />,
      ],
      "className": "row status-bar",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "amount": 124,
          "icon": "shopping-bag",
          "text": "New Orders",
          "theme": "blue",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "amount": 59,
          "icon": "comment",
          "text": "Comments",
          "theme": "yellow",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "amount": 9,
          "icon": "user",
          "text": "New Users",
          "theme": "green",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "amount": 17827,
          "icon": "eye",
          "text": "Page Views",
          "theme": "red",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <statusBarItem
            amount={124}
            icon="shopping-bag"
            text="New Orders"
            theme="blue"
          />,
          <statusBarItem
            amount={59}
            icon="comment"
            text="Comments"
            theme="yellow"
          />,
          <statusBarItem
            amount={9}
            icon="user"
            text="New Users"
            theme="green"
          />,
          <statusBarItem
            amount={17827}
            icon="eye"
            text="Page Views"
            theme="red"
          />,
        ],
        "className": "row status-bar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "amount": 124,
            "icon": "shopping-bag",
            "text": "New Orders",
            "theme": "blue",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "amount": 59,
            "icon": "comment",
            "text": "Comments",
            "theme": "yellow",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "amount": 9,
            "icon": "user",
            "text": "New Users",
            "theme": "green",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "amount": 17827,
            "icon": "eye",
            "text": "Page Views",
            "theme": "red",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
